app/src/main/java/com/sanaa/tudee_assistant/data/services/CategoryServiceImpl.kt

```
@@ -0,0 +1,36 @@
+package com.sanaa.tudee_assistant.data.services
+
+import com.sanaa.tudee_assistant.data.local.dao.CategoryDao
+import com.sanaa.tudee_assistant.data.local.mapper.toDomain
+import com.sanaa.tudee_assistant.data.local.mapper.toLocalDto
+import com.sanaa.tudee_assistant.domain.model.Category
+import com.sanaa.tudee_assistant.domain.service.CategoryService
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+class CategoryServiceImpl(
+    private val categoryDao: CategoryDao
+) : CategoryService {
+    override suspend fun getCategoryById(categoryId: Int): Category? {
+        return categoryDao.getCategoryById(categoryId)?.toDomain()
+    }
```

throw exception instead of returning null when item not found, also use try catch to map any exception that may occurs to domain exception